portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java

```diff
@@ -341,6 +341,11 @@ public DLFileEntry addFileEntry(
 			DLStoreUtil.addFile(dlStoreRequest, inputStream);
 		}
 
+		// FriendlyURL
+
+		_dlAppHelperLocalService.getEntryURL(
+			new LiferayFileEntry(dlFileEntry), serviceContext);
+
```

Comment:
The DLAppHelper should be used only from the DLApp services; the DLFileEntry* ones are too low level.